test/with_dummyserver/test_connectionpool.py

```diff
@@ -166,6 +166,68 @@ def test_connect_timeout(self):
         pool._put_conn(conn)
         self.assertRaises(ConnectTimeoutError, pool.request, 'GET', url, timeout=timeout)
 
+    def test_timeout_method_not_in_whitelist(self):
+        # Requests should time out when the method is not on the retry whitelist
+        block_event = Event()
+        ready_event = self.start_basic_handler(block_send=block_event, num=6)
+
+        # Pool-global timeout
+        timeout = Timeout(read=SHORT_TIMEOUT)
+        pool = HTTPConnectionPool(self.host, self.port, timeout=timeout, retries=True)
+
+        wait_for_socket(ready_event)
+        block_event.clear()
+        self.assertRaises(ReadTimeoutError, pool.request, 'POST', '/')
+        block_event.set() # Release request
+
+        # Request-specific timeouts should raise errors
+        pool = HTTPConnectionPool(self.host, self.port, timeout=LONG_TIMEOUT, retries=True)
+
+        conn = pool._get_conn()
+
+        wait_for_socket(ready_event)
+        now = time.time()
+        self.assertRaises(ReadTimeoutError, pool.request, 'POST', '/', timeout=timeout)
+        delta = time.time() - now
+
+        self.assertTrue(delta < LONG_TIMEOUT, "timeout was pool-level LONG_TIMEOUT rather than request-level SHORT_TIMEOUT")
+        block_event.set() # Release request
+
+        # Timeout int/float passed directly to request and _make_request should
+        # raise a request timeout
+        wait_for_socket(ready_event)
+        self.assertRaises(ReadTimeoutError, pool.request, 'POST', '/', timeout=SHORT_TIMEOUT)
+        block_event.set() # Release request
```

Comment:
@Lukasa Do you mean the whole test or this specific assert on line 199? The whole test is very similar to `test_timeout`, but it differs in that it passes `retries=True` to the pool class, and uses POST. Thus it tests that it does not retry on POST (which means it should raise a ReadTimeoutError). Do you think it should be more explicit, such as for instance having a specific method whitelist and Retry object?